src/convertToObject.js

```diff
@@ -6,7 +6,15 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map((styles) => styles.split(':').map((rule) => rule.trim()))
+    .filter(([key, val]) => key && val)
```

Comment:
The `filter` function is used correctly to remove entries with empty keys or values. However, ensure that the `map` function on line 11 correctly splits the string into key-value pairs before filtering.